rfhub2/cli/population.py

```diff
@@ -1,29 +1,36 @@
 from typing import Tuple, Dict, List
 import robot.libraries
 from robot.libdocpkg import LibraryDocumentation
-from requests import session, post
-from rfhub2.config import APP_INTERFACE, APP_PORT, BASIC_AUTH_USER, BASIC_AUTH_PASSWORD
+from .client import Client
 import os
 import re
 
 RESOURCE_PATTERNS = {".robot", ".txt", ".tsv", ".resource"}
 ALL_PATTERNS = (RESOURCE_PATTERNS | {".xml", ".py"})
 EXCLUDED_LIBRARIES = {"remote", "reserved", "dialogs", "dialogs_jy", "dialogs_py", "dialogs_ipy"}
-PROTOCOL = 'http://'
-API_V1 = 'api/v1'
 
 
-class LibraryPopulation(object):
+class AppPopulation(object):
 
-    def __init__(self, paths: Tuple[str, ...], no_installed_keywords: bool) -> None:
+    def __init__(self, app_interface: str, port: int, user: str, password: str,
+                 paths: Tuple[str, ...], no_installed_keywords: bool) -> None:
         self.paths = paths
-        self.auth = (BASIC_AUTH_USER, BASIC_AUTH_PASSWORD)
         self.no_installed_keywords = no_installed_keywords
+        self.client = Client(app_interface, port, user, password)
```

Comment:
Maybe pass client instance as an argument instead of initializing it inside? It will make unit testing easier.